website/docs/security/authentication.md

```diff
@@ -103,3 +116,41 @@ If you encounter any issues while authenticating with FerretDB, try setting the
 Please refer to your specific driver documentation on how to set this field.
 
 If this does not resolve your issue please file a bug report [here](https://github.com/FerretDB/FerretDB/issues/new?assignees=ferretdb-bot&labels=code%2Fbug%2Cnot+ready&projects=&template=bug.yml).
+
+## Experimental authentication
+
+Upon enabling the flag `--test-enable-new-auth` or the environment variable `TEST_ENABLE_NEW_AUTH`,
+FerretDB enables user management and authentication against the stored credentials in the `admin` database `system.users` collection.
+The users are managed using the `usersInfo`, `createUser`, `updateUser`, `dropUser` and `dropAllUsersFromDatabase` commands.
+The users may be created and authenticated using `SCRAM-SHA-256`, `SCRAM-SHA-1` and `PLAIN` mechanisms.
+The `SCRAM` mechanisms are the algorithms used to store hashed passwords. FerretDB also stores hashed password using `PLAIN` mechanism.
+
+Please note that FerretDB does not support authorization yet, so created users currently have access to all databases.
+
+To connect to FerretDB, if a client use `mongodb://user:pass@ferretdb:27018/ferretdb?tls=true&authMechanism=SCRAM-SHA-256`,
+`user` is authenticated against its credential stored in the `admin.system.users` collection using `SCRAM-SHA-256` authentication mechanism.
+
+Before the first user is created in FerretDB, the credentials provided in the MongoDB connection string are used to connect directly to the PostgreSQL backend via passthrough.
+For instance, when the `admin.system.users` collection is empty and
+a client connects as `mongodb://pguser:pgpass@ferretdb:27018/ferretdb?tls=true&authMechanism=PLAIN`,
+it uses `pguser` to connect to the postgreSQL backend.
+When authenticating directly to the backend, only `PLAIN` mechanism is supported.
+Please note this exception no longer applies once the first user is created.
+
+When usernames and passwords are transferred in plain text or on an unsecured network,
```

Comment:
Ok, _when_ passwords are transferred in plain text? We don't actually say that.

And usernames are always transferred in plain text and that's fine.

Reply:
Updated